### ISSUE TITLE
Retention Application : Delete change_events, activity threads, versions based on admin retention policies

### ISSUE THREAD

**Issue by aalmeida-st** [open-metadata/OpenMetadata#15600]
**Is your feature request related to a problem? Please describe.**

We noticed that the table `change_event` is growing and is consuming the majority of the space in the database.
This table, as far as we understood, saves data related to the events that happen in the application, however, it looks the data saved there is not used in any place on the application.
These events are saved in json objects which can be large objects.

**Describe the solution you'd like**
It would be important if we could reduce the history that is saved on this table, maybe some parameter that can be passed, or keep the last x days.
Reducing this table will mean reducing storage costs which is very important.

**Describe alternatives you've considered**

**Additional context**

**Comment by BinhPhamQuang** [comment]
I created this cronjob running hourly to remove old data. This solution worked well for me:
```
export MYDB=postgresql://:$DB_USER_PASSWORD@host/db;
SCHEMA=openmetadata;
BOTS="('ingestion-bot','DataInsightsApplicationBot')"; 
echo "DELETE FROM $SCHEMA.thread_entity WHERE createdby in $BOTS and type = 'Conversation'; DELETE FROM $SCHEMA.change_event where username in $BOTS;" > script.sql; cat script.sql && psql -Atx $MYDB -f script.sql
```

**Comment by BinhPhamQuang** [comment]
> > I created this cronjob running hourly to remove old data. This solution worked well for me:
> > ```
> > export MYDB=postgresql://:$DB_USER_PASSWORD@host/db;
> > SCHEMA=openmetadata;
> > BOTS="('ingestion-bot','DataInsightsApplicationBot')"; 
> > echo "DELETE FROM $SCHEMA.thread_entity WHERE createdby in $BOTS and type = 'Conversation'; DELETE FROM $SCHEMA.change_event where username in $BOTS;" > script.sql; cat script.sql && psql -Atx $MYDB -f script.sql
> > ```
> 
> Thank you @BinhPhamQuang is what we will do also. But I was wondering if could be something that we could configure on the OM side.

Yes, I agree with you. I think OM side should have retention feature.  It shouldn't be limited to just the change_event table. In my opinion, the version table and profile also tend to accumulate substantial data on a daily basis.
